src/metalperformanceshadersgraph.cs

```diff
@@ -0,0 +1,1840 @@
+using System;
+using CoreGraphics;
+using Foundation;
+using Metal;
+using MetalPerformanceShaders;
+using ObjCRuntime;
+
+#if !NET
+using NativeHandle = System.IntPtr;
+#endif
+
+using MPSGraphTensorDataDictionary = Foundation.NSDictionary<MetalPerformanceShadersGraph.MPSGraphTensor, MetalPerformanceShadersGraph.MPSGraphTensorData>;
+using MPSGraphTensorShapedTypeDictionary = Foundation.NSDictionary<MetalPerformanceShadersGraph.MPSGraphTensor, MetalPerformanceShadersGraph.MPSGraphShapedType>;
+
+
+// MPSGraphType was introduced in iOS 15 (macOS 12) and became the base class for
+// MPSGraphShapedType which exists in iOS 14.
+// In iOS 14, MPSGraphShapedType inherits directly from NSObject.
+// To keep compatibility, MPSGraphType is type-erased to just NSObject
+// and implementers have to add NSCopying.
+// @interface MPSGraphType : NSObject
+// [TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+// [BaseType (typeof(NSObject))]
+// interface MPSGraphType : NSCopying
+// {
+// }
+using MPSGraphType = Foundation.NSObject;
```

Comment:
I think it's better to modify the BaseType attribute for `MPSGraphShapedType` to be `NSObject`, and then bind `MPSGraphType` (like you've commented it out here).

As it stands today, your solution works fine, but if Apple decides to add API to `MPSGraphType` in the future, and we've bound methods with `NSObject` parameters instead of the correct `MPSGraphType` type, then it becomes somewhat cumbersome to fix those methods to have the right parameter types.

Reply:
Does Apple do this often? It's very mean of them. :-) I'll add the type back in.

Reply:
I ran into a snag... I'm worried I won't be able to pass a `MPSGraphShapedType` to `Specialize` if I reify `MPSGraphType`.

This is the only usage of MPSGraphType (introduced in iOS 15):

```csharp
// -(void)specializeWithDevice:(MPSGraphDevice * _Nullable)device inputTypes:(NSArray<MPSGraphType *> * _Nonnull)inputTypes compilationDescriptor:(MPSGraphCompilationDescriptor * _Nullable)compilationDescriptor;
[Export ("specializeWithDevice:inputTypes:compilationDescriptor:")]
void Specialize ([NullAllowed] MPSGraphDevice device, MPSGraphType[] inputTypes, [NullAllowed] MPSGraphCompilationDescriptor compilationDescriptor);
```

`MPSGraphShapedType` is used in `Compile` as a member of a dictionary - also introduced in iOS 15.

May I propose an alternate solution? Since this type family is only used in APIs marked iOS 15, I think I should just change `MPSGraphShapedType` to also be iOS 15-only and then do the inheritance right. (That is, ignore iOS 14).

Reply:
I added `MPSGraphType` in https://github.com/xamarin/xamarin-macios/pull/14303/commits/3967104440388c75336e81011dc546c1ad68f791

Reply:
> May I propose an alternate solution? Since this type family is only used in APIs marked iOS 15, I think I should just change MPSGraphShapedType to also be iOS 15-only and then do the inheritance right - including MPSGraphType as a full type.

Yes, I think that's fine.